lib/livebook/storage/ets.ex

```diff
@@ -72,10 +76,18 @@ defmodule Livebook.Storage.Ets do
   end
 
   @impl GenServer
-  def init(_opts) do
-    table = :ets.new(@table_name, [:named_table, :protected, :duplicate_bag])
+  def init(opts) do
+    # enable passing table name for testing purposes
+    table_name = Keyword.get(opts, :table_name, @table_name)
```

Comment:
Operations like `fetch_key` don't have access to state and always use `@table_name`, so this doesn't fully work (even if that's just for tests it should) :<

Reply:
It works if you use the internals ( ͡° ͜ʖ ͡°)

Reply:
Nono, my point is that it works for some part of the functionality, the one relevant for tests, but having changed the table name there's no way to use `fetch` properly (and if we have the option, we should make sure it does :)).

Reply:
How about making that option available only for `MIX_ENV=test`? I know that it sounds hacky but that is the nature of singletons :(

Reply:
I dont think you need a separate table. Writing to the global table is fine as long as you use a distinct keys :)

Reply:
I don't want to run into some race conditions caused by using the global table. I still need to reload it to check if it catched changes from file correctly.

Reply:
@Qizot right, the discussion is spread out over two comments, so maybe there is something getting list. I think the implementation could:

1. no longer use a named table
2. persist after every write

This means that you can observe writes to the file by having the test itself call `:ets.file2tab` and load a separate instance of the table.

In this approach, you don't need any "backdoor" in the ets process. It always write to disk and you load the disk contents from the test. The writing itself will be done through the Livebook.Storage API, something that we already do in tests and it works fine, because we make sure that each test uses a distinct key. Given we don't have an API for reading all entries, it is fine, there is no way we can read or access the keys for an entry we don't know about!

Reply:
Done 📦